htdocs/kernel/block.php

```
@@ -368,6 +358,65 @@ public function getContent($format = 's', $c_type = 'T')
+        if (!(defined('XOOPS_ALLOW_PHP_BLOCKS') && constant('XOOPS_ALLOW_PHP_BLOCKS') === true)) {
+            $logger = XoopsLogger::getInstance();
+            $logger->addDeprecated(
+                'Legacy PHP block detected. Migrate to file-based format '
+                . '(filename.php|function_name in custom_blocks/) or set '
+                . 'XOOPS_ALLOW_PHP_BLOCKS to true in mainfile.php.'
+            );
+            return '';
+        }
+
+        ob_start();
+        echo eval($raw);
```

PHPMD (2.15.0)

[error] The method executePhpBlock() contains an eval expression. (undefined)

(EvalExpression)